src/main/java/com/labsynch/labseer/service/StandardizationServiceImpl.java

```diff
@@ -457,52 +526,91 @@ public int restandardizeParentStructures(List<Long> parentIds)
 		int totalCount = parentIds.size();
 		logger.info("number of parents to restandardize: " + totalCount);
 		float percent = 0;
-		int p = 1;
+		int p = 0;
 		float previousPercent = percent;
 		previousPercent = percent;
 
 		Session session = Parent.entityManager().unwrap(Session.class);
 		Long startTime = new Date().getTime();
 		Long currentTime = new Date().getTime();
 		
-		for (Long parentId : parentIds) {
-			parent = Parent.findParent(parentId);
-			lots = Lot.findLotsByParent(parent).getResultList();
-			lot = lots.get(0);
 
-			// Try getting the as drawn structure first if we have it
-			if (lots.size() > 0 && lot.getAsDrawnStruct() != null) {
-				originalStructure = lot.getAsDrawnStruct();
-			} else {
-				logger.warn("Did not find the asDrawnStruct for parent: " + parentId + "  " + parent.getCorpName());
-				originalStructure = parent.getMolStructure();
+		// Split parent ids into groups of batchSize
+		List<List<Long>> parentIdGroups = new ArrayList<List<Long>>();
+		List<Long> parentIdGroup = new ArrayList<Long>();
+		for (Long parentId : parentIds) {
+			parentIdGroup.add(parentId);
+			if (parentIdGroup.size() == batchSize) {
+				parentIdGroups.add(parentIdGroup);
+				parentIdGroup = new ArrayList<Long>();
 			}
+		}
 
-			// We standardize the structure first
-			standardizedMol = chemStructureService.standardizeStructure(originalStructure);
+		// Do a bulk standardization
+		for (List<Long> pIdGroup : parentIdGroups) {
+			logger.info("Starting batch of " + pIdGroup.size() + " parents");
+
+			// Create standardization hashmap
+			HashMap<String, String> inputStructures = new HashMap<String, String>();
+			HashMap<Long, Parent> parents = new HashMap<Long, Parent>();
+			for(Long parentId : pIdGroup) {
+				parent = Parent.findParent(parentId);
+				parents.put(parentId, parent);
+				inputStructures.put(parentId.toString(parentId), parent.getMolStructure());
+			}
 
-			// Now we update the parent structure
-			Boolean success = updateStructure(standardizedMol, parent.getCdId());
+			// Do standardization
+			logger.debug("Starting standardization of " + inputStructures.size() + " compounds");
+			// Start timer
+			long standardizationStart = new Date().getTime();
+			HashMap<String, CmpdRegMolecule> standardizationResults = chemStructureService.standardizeStructures(inputStructures);
+			long standardizationEnd = new Date().getTime();
+			// Convert the ms time to seconds
+			long standardizationTime = (standardizationEnd - standardizationStart) / 1000;
+			logger.debug("Standardization took " + standardizationTime + " seconds");
```

Comment:
* helper function to split into groups (lines 538 - 547)
Thanks, I did this one.
* helper function that standardizes a parentIdGroup (lines 551 to 570)
This a little awkward because that block creates 2 outputs, an array of Parents and a standardized structure hashmap. I am a little reluctant to loose the efficiency of this single loop and its also a little awkward to return 2 things from a single function.  
* helper function to get original structure from a parent (lines 575-584)
I'm going to refactor this soon anyway so going to skip for now.

Reply:
Makes sense, fine by me to skip refactoring the second. case